fuzz: oss-fuzz coverage build is failing

It first failed on the 25th (https://oss-fuzz-build-logs.storage.googleapis.com/log-7da28be4-dcf5-4e23-a6ca-153f67ad2f60.txt):
```bash
Step #7: [0m[0;31mwarning: The file '/usr/evmap.c' isn't covered.
Step #7: [0m[0;31merror: /workspace/out/libfuzzer-coverage-x86_64/usr/arc4random.c: No such file or directory
Step #7: [0m[0;31mwarning: The file '/usr/arc4random.c' isn't covered.
Step #7: [0m[0;31merror: /workspace/out/libfuzzer-coverage-x86_64/usr/buffer.c: No such file or directory
Step #7: [0m[0;31mwarning: The file '/usr/buffer.c' isn't covered.
Step #7: [0m[0;31merror: /workspace/out/libfuzzer-coverage-x86_64/usr/util-internal.h: No such file or directory
Step #7: [0m[0;31mwarning: The file '/usr/util-internal.h' isn't covered.
Step #7: [0m[2025-01-31 07:01:44,276 DEBUG] Finished generating per-file code coverage summary.
Step #7: [2025-01-31 07:01:44,277 DEBUG] Generating file view html index file as: "/workspace/out/libfuzzer-coverage-x86_64/report/linux/file_view_index.html".
Step #7: Traceback (most recent call last):
Step #7:   File "/opt/code_coverage/coverage_utils.py", line 829, in <module>
Step #7:     sys.exit(Main())
Step #7:   File "/opt/code_coverage/coverage_utils.py", line 823, in Main
Step #7:     return _CmdPostProcess(args)
Step #7:   File "/opt/code_coverage/coverage_utils.py", line 780, in _CmdPostProcess
Step #7:     processor.PrepareHtmlReport()
Step #7:   File "/opt/code_coverage/coverage_utils.py", line 577, in PrepareHtmlReport
Step #7:     self.GenerateFileViewHtmlIndexFile(per_file_coverage_summary,
Step #7:   File "/opt/code_coverage/coverage_utils.py", line 450, in GenerateFileViewHtmlIndexFile
Step #7:     self.GetCoverageHtmlReportPathForFile(file_path),
Step #7:   File "/opt/code_coverage/coverage_utils.py", line 422, in GetCoverageHtmlReportPathForFile
Step #7:     assert os.path.isfile(
Step #7: AssertionError: "/usr/arc4random.c" is not a file.
Step #7: ********************************************************************************
Step #7: Code coverage report generation failed.
Step #7: To reproduce, run:
Step #7: python infra/helper.py build_image bitcoin-core
Step #7: python infra/helper.py build_fuzzers --sanitizer coverage bitcoin-core
Step #7: python infra/helper.py coverage bitcoin-core
Step #7: ********************************************************************************
Finished Step #7
ERROR
ERROR: build step 7 "gcr.io/oss-fuzz-base/base-runner" failed: step exited with non-zero status: 1
```

See also: https://issues.oss-fuzz.com/issues/391974928.

--- a/depends/packages/capnp.mk
+++ b/depends/packages/capnp.mk
@@ -9,7 +9,7 @@ define $(package)_set_vars :=
   $(package)_config_opts := -DBUILD_TESTING=OFF
   $(package)_config_opts += -DWITH_OPENSSL=OFF
   $(package)_config_opts += -DWITH_ZLIB=OFF
-  $(package)_cxxflags += -ffile-prefix-map=$$($(package)_extract_dir)=/usr
+  $(package)_cxxflags += -fdebug-prefix-map=$($(package)_extract_dir)=/usr -fmacro-prefix-map=$($(package)_extract_dir)=/usr
 endef
 
 define $(package)_config_cmds

--- a/depends/packages/libevent.mk
+++ b/depends/packages/libevent.mk
@@ -13,7 +13,7 @@ define $(package)_set_vars
   $(package)_config_opts=-DCMAKE_BUILD_TYPE=None -DEVENT__DISABLE_BENCHMARK=ON -DEVENT__DISABLE_OPENSSL=ON
   $(package)_config_opts+=-DEVENT__DISABLE_SAMPLES=ON -DEVENT__DISABLE_REGRESS=ON
   $(package)_config_opts+=-DEVENT__DISABLE_TESTS=ON -DEVENT__LIBRARY_TYPE=STATIC
-  $(package)_cflags += -ffile-prefix-map=$($(package)_extract_dir)=/usr
+  $(package)_cflags += -fdebug-prefix-map=$($(package)_extract_dir)=/usr -fmacro-prefix-map=$($(package)_extract_dir)=/usr
   $(package)_cppflags += -D_GNU_SOURCE
   $(package)_cppflags_mingw32=-D_WIN32_WINNT=0x0A00
 

--- a/depends/packages/libmultiprocess.mk
+++ b/depends/packages/libmultiprocess.mk
@@ -13,7 +13,7 @@ ifneq ($(host),$(build))
 $(package)_config_opts := -DCAPNP_EXECUTABLE="$$(native_capnp_prefixbin)/capnp"
 $(package)_config_opts += -DCAPNPC_CXX_EXECUTABLE="$$(native_capnp_prefixbin)/capnpc-c++"
 endif
-$(package)_cxxflags += -ffile-prefix-map=$$($(package)_extract_dir)=/usr
+$(package)_cxxflags += -fdebug-prefix-map=$($(package)_extract_dir)=/usr -fmacro-prefix-map=$($(package)_extract_dir)=/usr
 endef
 
 define $(package)_config_cmds

--- a/depends/packages/zeromq.mk
+++ b/depends/packages/zeromq.mk
@@ -17,7 +17,7 @@ define $(package)_set_vars
   $(package)_config_opts += -DWITH_LIBBSD=OFF -DENABLE_CURVE=OFF -DENABLE_CPACK=OFF
   $(package)_config_opts += -DBUILD_SHARED=OFF -DBUILD_TESTS=OFF -DZMQ_BUILD_TESTS=OFF
   $(package)_config_opts += -DENABLE_DRAFTS=OFF -DZMQ_BUILD_TESTS=OFF
-  $(package)_cxxflags += -ffile-prefix-map=$($(package)_extract_dir)=/usr
+  $(package)_cxxflags += -fdebug-prefix-map=$($(package)_extract_dir)=/usr -fmacro-prefix-map=$($(package)_extract_dir)=/usr
   $(package)_config_opts_mingw32 += -DZMQ_WIN32_WINNT=0x0A00 -DZMQ_HAVE_IPC=OFF
 endef
 

